modules/web/js/ballerina/components/globals-expanded.jsx

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import './globals-expanded.css'
+import ImageUtil from './image-util';
+import PropTypes from 'prop-types';
+import SuggestionsText from './suggestions-text';
+import GlobalItem from './global-item';
+import EditableText from './editable-text';
+import BallerinaEnvironment from '../env/environment';
+
+export default class GlobalExpanded extends React.Component {
+    constructor() {
+        super();
+        this.state = {
+            editing: false,
+            value: '',
+        }
+        this.handleAddGlobalClick = this.handleAddGlobalClick.bind(this);
+        this.handleAddGlobalBlur = this.handleAddGlobalBlur.bind(this);
+        this.onChange = this.onChange.bind(this);
+        this.onKeyDown = this.onKeyDown.bind(this);
+    }
+
+    handleAddGlobalClick() {
+        this.setState({editing: true});
+    }
+
+    handleAddGlobalBlur() {
+        this.setState({editing: false});
+        this.props.onAddInputBlur && this.props.onAddInputBlur(this.state.value);
+    }
+
+    onChange(e) {
+        this.setState({value: e.target.value});
+    }
+
+    onKeyDown(e) {
+        if(e.keyCode === 13) {
+            this.props.onAddNewValue(this.state.value);
+            this.setState({
+                value: ''
+            });
+        }
+    }
+
+    render() {
+        const bBox = this.props.bBox;
+        const globalHeight = 30;
+        const globalInputHeight = 40;
+        const globalDeclarationWidth = 310;
+        const leftPadding = 10;
+        const topGutter = 10;
+        const topBarHeight = 25;
+        const iconSize = 20;
```

Comment:
Since this is a separate component, shall we move these constant values to our defaults file